packages/back-end/src/services/features.ts

```diff
@@ -242,7 +241,11 @@ export async function refreshSDKPayloadCache(
 
   // Purge CDN if used
   // Do this before firing webhooks in case a webhook tries fetching the latest payload from the CDN
-  await purgeCDNCache(organization.id, payloadKeys);
+  // Only purge the specific payloads that are affected
+  const surrogateKeys = payloadKeys.map((k) =>
+    getSurrogateKey(organization.id, k.project, k.environment)
+  );
```

Comment:
We could move this map logic into the `getSurrogateKey` function and give it a clearer name

```ts
export function getSurrogateKeysFromSDKPayloadKeys(
  orgId: string,
  payloadKeys: SDKPayloadKey[]
): string[] {
  ...
}
```